easybuild/easyconfigs/q/QuTiP/QuTiP-4.7.1-foss-2021b.eb

```
@@ -0,0 +1,33 @@
+easyblock = 'PythonPackage'
+
+name = 'QuTiP'
+version = '4.7.1'
+
+homepage = 'http://qutip.org'
+description = """QuTiP is open-source software for simulating the dynamics of open quantum systems."""
+
+toolchain = {'name': 'foss', 'version': '2021b'}
+
+source_urls = ['http://github.com/%(namelower)s/%(namelower)s/archive']
+sources = ['v%(version)s.tar.gz']
+checksums = ['9b40ce42e881061c3fe4f0484bc049d4dc9afc1931826836231313c4f8ab321d']
+
+dependencies = [
+    ('Python', '3.9.6'),
+    ('matplotlib', '3.4.3'),
+]
+
+use_pip = True
+download_dep_fail = True
+sanity_pip_check = True
+
+# verify installation (takes a while), best performed with > 10 CPUs
+# see also http://qutip.org/docs/latest/installation.html#verifying-the-installation
+sanity_check_commands = ["python -c 'import qutip.testing as qt; qt.run()'"]
```

There are problems with this approach that they suggest, and, worth noting, this is not what they themselves do in their continuous integration; they just run pytest like normal (https://github.com/qutip/qutip/blob/master/.github/workflows/tests.yml#L162) 

So, we can't have this in sanity check, it runs each time you rebuild a module, can't be opted out of, and sanity checks should always be extremely quick.
I also strongly suspect that this wouldn't exit nonzero ever anyway, so this wouldn't catch any errors.

So, basically, what is needed here is something like
```suggestion
testinstall = True
runtest = 'pytest qutip/tests'
```

but i'm having some issues here w.r.t to the existence of the build dir sources interfering with the installed version, making trivial "import qutip.version" fail in pytest.

I could make it run by brute force moving qutip/tests directory out, but there should definitely be some nicer way of doing this than trying to hack that into pretestopts..

---

I noticed the test suite also creates a bunch of plots in the working dir, so, that puts yet more limitations on where this can run as to not pollute the installation with this.

If we could cd into the testinstall dir, `"pytest qutip/tests"`  works great, but, this isn't exposed in the easyblock :( 

So, best i have right now is something like
`runtest = 'rm qutip/tests/__init__,py; pytest qutip/tests'`

as that __init__.py file makes pytest act like it's part of the module, and adds the sys path to the interfering src dir, which, since it lacks all the compiled pyx files etc. will immediatley fail.
Just removing the (empty) __init__ file makes it possible to run this from the build dir.

Though, I am seeing some minor failures, some of which i think needs a few --ignore flags to pytest as i suspect it's optional stuff.

---

This is indeed ... unfortunate.

Honestly? After much tinkering, I took the example from `QuTiP-4.3.1-intel-2018b-Python-3.6.6.eb`. And the remark about the number of CPUs? At first, I tried on our login with `--parallel=2`. Not clever. Then in jobcontext I observed thread usages between 4-20. No scaling test was done for the test.

So, how do we proceed here? I cannot put too much time in this.